lib/mix/tasks/hex.ex

```diff
@@ -43,18 +43,33 @@ defmodule Mix.Tasks.Hex do
 
   defp build_task_doc_list(modules) do
     Enum.reduce(modules, {[], 0}, fn module, {docs, max} ->
-      if doc = Mix.Task.shortdoc(module) do
-        task = "mix " <> Mix.Task.task_name(module)
-        {[{task, doc} | docs], max(byte_size(task), max)}
+      task = "mix " <> Mix.Task.task_name(module)
+      taskList = if doc = Mix.Task.shortdoc(module) do
+        [{task, doc}]
       else
+        []
+      end ++ if Keyword.has_key?(module.__info__(:functions), :subtasks) do
+        module.subtasks() |> Enum.map(fn {subtask, docs} -> {"#{task} #{subtask}", docs} end)
```

Comment:
We usually avoid single pipes

```suggestion
        Enum.map(module.subtasks(), fn {subtask, docs} -> {"#{task} #{subtask}", docs} end)
```